ports/stm32/subghz_radio.c

```diff
@@ -0,0 +1,193 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2022 Angus Gratton
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include "py/runtime.h"
+#include "subghz_radio.h"
+#include "irq.h"
+#include "spi.h"
+
+#if MICROPY_HW_ENABLE_SUBGHZ_RADIO
+
+extern mp_obj_t machine_hard_spi_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *all_args);
+
+typedef struct _machine_subghz_radio_obj_t {
+    mp_obj_base_t base;
+    mp_obj_t spi;
+} machine_subghz_radio_obj_t;
+
+STATIC machine_subghz_radio_obj_t machine_subghz_radio_obj = {{&machine_subghz_radio_type}, NULL};
+
+void handle_subghz_radio_irq() {
+    /* Level-triggered interrupts means the interrupt has to be cleared before this function returns.
+
+       Rather than writing to SUBGHZ SPI in Interrupt Context to clear the interrupt, disable the IRQ
+       and rely on the radio driver to call radio.irq_en(True) to re-enable when needed.
+    */
```

Comment:
C++ style comments

Reply:
I finally have these highlighted as errors in emacs!